cylc/flow/run_modes/skip.py

```diff
@@ -0,0 +1,173 @@
+    task_job_mgr: 'TaskJobManager',
+    itask: 'TaskProxy',
+    rtconfig: Dict[str, Any],
+    workflow: str,
+    now: Tuple[float, str]
+) -> 'Literal[True]':
+    """Submit a task in skip mode.
+
+    Returns:
+        True - indicating that TaskJobManager need take no further action.
+    """
+    itask.summary['started_time'] = now[0]
+    # TODO - do we need this? I don't thing so?
+    task_job_mgr._set_retry_timers(itask, rtconfig)
+    itask.waiting_on_job_prep = False
+    itask.submit_num += 1
+
+    itask.platform = get_platform()
+    itask.platform['name'] = RunMode.SKIP
+    itask.summary['job_runner_name'] = RunMode.SKIP
+    itask.tdef.run_mode = RunMode.SKIP
+    task_job_mgr.task_events_mgr.process_message(
+        itask, INFO, TASK_OUTPUT_SUBMITTED,
+    )
+    task_job_mgr.workflow_db_mgr.put_insert_task_jobs(
+        itask, {
+            'time_submit': now[1],
+            'try_num': itask.get_try_num(),
+        }
+    )
+    for output in process_outputs(itask):
+        task_job_mgr.task_events_mgr.process_message(itask, INFO, output)
+
+    return True
+
+
+def process_outputs(itask: 'TaskProxy') -> List[str]:
+    """Process Skip Mode Outputs:
+
+    * By default, all required outputs will be generated plus succeeded
+      if success is optional.
+    * The outputs submitted and started are always produced and do not
+      need to be defined in outputs.
+    * If outputs is specified and does not include either
+      succeeded or failed then succeeded will be produced.
+
+    Return:
+        A list of outputs to emit.
+    """
+    result: List = []
+    conf_outputs = itask.tdef.rtconfig['skip']['outputs']
+
+    # Remove started or submitted from our list of outputs:
+    for out in get_unecessary_outputs(conf_outputs):
+        conf_outputs.remove(out)
+
+    # Submitted will always be produced:
+    # (Task events manager handles started automatically for ghost modes)
+    result.append(TASK_OUTPUT_SUBMITTED)
+
+    # Send the rest of our outputs, unless they are succeed or failed,
+    # which we hold back, to prevent warnings about pre-requisites being
+    # unmet being shown because a "finished" output happens to come first.
+    for message in itask.state.outputs.iter_required_messages():
+        trigger = itask.state.outputs._message_to_trigger[message]
+        # Send message unless it be succeeded/failed.
+        if trigger in [TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED]:
+            continue
+        if not conf_outputs or trigger in conf_outputs:
+            result.append(message)
+
+    # Send succeeded/failed last.
+    if TASK_OUTPUT_FAILED in conf_outputs:
+        result.append(TASK_OUTPUT_FAILED)
+    else:
+        result.append(TASK_OUTPUT_SUCCEEDED)
+
+    return result
+
+
+def check_task_skip_config(name: str, tdef: 'TaskDef') -> None:
+    """Ensure that skip mode configurations are sane at validation time:
+
+    Args:
+        name: of task
+        tdef: of task
```

Comment:
Why not use `tdef.name` instead of a separate `name` arg?

Reply:
👍🏼